packages/core/src/components/Lightbox/index.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Sheet from '../Sheet';
+import LightboxImage, { LightboxImageProps } from './LightboxImage';
+import LightboxHeader from './LightboxHeader';
+
+export type LightboxProps = {
+  /** Images to show. */
+  images: LightboxImageProps[];
+  /** Image start index. */
+  startIndex?: number;
+  /** Show zoom controls. */
+  showZoomControls?: boolean;
+  /** Show rotate controls. */
+  showRotateControls?: boolean;
+  /** Function to run on close. */
+  onClose: () => void;
+};
+
+export type LightboxState = {
+  activeIndex: number;
+  hideAside: boolean;
+  scale: number;
+  rotation: number;
+};
+
+export default class Lightbox extends React.PureComponent<LightboxProps, LightboxState> {
+  static defaultProps = {
+    startIndex: 0,
+  };
+
+  state = {
+    activeIndex: 0,
+    hideAside: false,
+    scale: 1,
+    rotation: 0,
+  };
+
+  preloadedUrls: Map<string, boolean | HTMLImageElement>;
```

Comment:
`preloadedUrls = new Map<string, boolean | HTMLImageElement>();` and then you can remove the constructor.